File: backend-springboot/src/main/java/br/com/portalproposta/backend/service/ConfiguracaoService.java
package br.com.portalproposta.backend.service;

import br.com.portalproposta.backend.exception.AppBadRequestException;
import br.com.portalproposta.backend.model.Configuracao;
import br.com.portalproposta.backend.repository.ConfiguracaoRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.Locale;

@Service
public class ConfiguracaoService {

    private final ConfiguracaoRepository configuracaoRepository;

    @Autowired
    public ConfiguracaoService(ConfiguracaoRepository configuracaoRepository) {
        this.configuracaoRepository = configuracaoRepository;
    }

    public double getCommissionRate() {
        return configuracaoRepository.findById("taxa_corretagem")
                .map(conf -> {
                    try {
                        return Double.parseDouble(conf.getValor());
                    } catch (NumberFormatException e) {
                        return 5.0;
                    }
                })
                .orElse(5.0);
    }

    @Transactional
    public void updateCommissionRate(double rate) {
        if (rate < 0 || rate > 100) {
            throw new AppBadRequestException("taxa de corretagem inválida (deve ser entre 0% e 100%)");
        }
        String val = String.format(Locale.US, "%.2f", rate);
        setSetting("taxa_corretagem", val);
    }

    public void verifySupervisorPassword(String inputPassword) {
        String supervisorPassword = configuracaoRepository.findById("senha_supervisor")
                .map(Configuracao::getValor)
                .orElse("admin123");

        if (!supervisorPassword.equals(inputPassword)) {
            throw new AppBadRequestException("senha do supervisor incorreta");
        }
    }

    @Transactional
    public void updateSupervisorPassword(String newPassword) {
        if (newPassword == null || newPassword.trim().length() < 4) {
            throw new AppBadRequestException("a senha do supervisor deve ter no mínimo 4 caracteres");
        }
        setSetting("senha_supervisor", newPassword.trim());
    }

    @Transactional
    public void setSetting(String chave, String valor) {
        Configuracao conf = configuracaoRepository.findById(chave)
                .orElseGet(() -> Configuracao.builder().chave(chave).build());
        conf.setValor(valor);
        configuracaoRepository.save(conf);
    }
}

File: backend/Services/ConfiguracaoService.cs
using Microsoft.EntityFrameworkCore;
using PortalProposta.Data;
using PortalProposta.Models;
using System.Globalization;

namespace PortalProposta.Services;

public class ConfiguracaoService(AppDbContext context)
{
    public async Task<double> GetCommissionRateAsync()
    {
        var conf = await context.Configuracoes.FirstOrDefaultAsync(c => c.Chave == "taxa_corretagem");
        return conf is not null && double.TryParse(conf.Valor, NumberStyles.Any, CultureInfo.InvariantCulture, out var rate)
            ? rate
            : 5.0;
    }

    public async Task UpdateCommissionRateAsync(double rate)
    {
        if (rate < 0 || rate > 100)
            throw new AppBadRequestException("taxa de corretagem inválida (deve ser entre 0% e 100%)");

        await SetSettingAsync("taxa_corretagem", rate.ToString("F2", CultureInfo.InvariantCulture));
    }

    public async Task VerifySupervisorPasswordAsync(string inputPassword)
    {
        var conf = await context.Configuracoes.FirstOrDefaultAsync(c => c.Chave == "senha_supervisor");
        var actualPassword = conf?.Valor ?? "admin123";

        if (actualPassword != inputPassword)
            throw new AppBadRequestException("senha do supervisor incorreta");
    }

    public async Task UpdateSupervisorPasswordAsync(string newPassword)
    {
        if (string.IsNullOrWhiteSpace(newPassword) || newPassword.Trim().Length < 4)
            throw new AppBadRequestException("a senha do supervisor deve ter no mínimo 4 caracteres");

        await SetSettingAsync("senha_supervisor", newPassword.Trim());
    }

    private async Task SetSettingAsync(string chave, string valor)
    {
        var conf = await context.Configuracoes.FirstOrDefaultAsync(c => c.Chave == chave);
        if (conf is null)
        {
            context.Configuracoes.Add(new() { Chave = chave, Valor = valor });
        }
        else
        {
            conf.Valor = valor;
        }
        await context.SaveChangesAsync();
    }
}
